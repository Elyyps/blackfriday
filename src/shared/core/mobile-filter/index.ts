export * from "./mobile-filter.component";
export * from "./dummy-data";
