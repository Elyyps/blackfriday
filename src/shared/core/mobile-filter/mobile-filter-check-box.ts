export interface IMobileFilterCheckBox {
  checked: boolean;
  title: string;
}

