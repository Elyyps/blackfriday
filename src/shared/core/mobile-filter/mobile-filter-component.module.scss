.filter-bar {
  background-color: #f0f0f0;
  color: #1c1c1c;
  display: inline-flex;
  width: 100%;
  justify-content: space-between;
  font-size: 15px;
  font-weight: 600;
  box-sizing: border-box;
  padding: 12px 16px;
  align-items: center;
  cursor: pointer;
}
.mobile-filter {
  margin: auto;
  height: 100%;
  position: relative;
  box-sizing: border-box;
  &__footer {
    border-top: #e0e0e0 1px solid;
    padding: 16px !important;
    position: fixed;
    bottom: 0;
    z-index: 100;
    left: 0;
    right: 0;
    position: fixed;
    box-sizing: border-box;
    background: #f0f0f0;
  }
  &__list {
    padding: 16px;
    &__header {
      box-sizing: border-box;
      margin-bottom: 8px;
      background-color: #f0f0f0;
      height: 40px;
      width: 100%;
      font-weight: 600;
    }
    &__item {
      padding: 16px 0;
      border-bottom: 1px solid #e0e0e0;
      * {
        font-size: 16px;
        font-weight: 600;
      }
    }
  }
  &__header {
    min-height: 44px;
    height: 44px;
    width: 100%;
    text-align: center;
    background-color: #1c1c1c;
    font-weight: 600;
    a {
      color: white;
      font-weight: 500;
      text-decoration: underline;
    }
    &__title {
      display: inline-flex;
      box-sizing: border-box;
      span {
        margin-right: 11px;
      }

      svg {
        transform: rotate(90deg);
        margin-top: 2px;
      }
    }
    &__clear {
      position: absolute;
      right: 16px;

      text-decoration: underline !important;
    }
    &__nav-button {
      position: absolute;
      left: 16px;
      cursor: pointer;
      svg {
        margin: 0;
        &:hover,
        &:active {
          outline: none;
        }
      }
    }
    min-height: 44px;
    box-sizing: border-box;
    line-height: 1;
    display: flex;
    justify-content: center;
    padding: 12px 16px;
    background-color: $gray-dark;
    color: $global-inverse-color;
    position: relative;
    align-items: center;
  }
}
