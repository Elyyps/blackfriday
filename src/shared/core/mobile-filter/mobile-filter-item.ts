export interface IMobileFilterItem {
  hasSearchBar: boolean;
  isSingleSelection?: boolean;
  items: string[];
  searchBarPlaceholder?: string;
  selectedItems: string[];
  title: string;
}
