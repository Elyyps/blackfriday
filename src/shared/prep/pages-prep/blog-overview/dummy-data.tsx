import Oasis from "@assets/oasis.jpg";
import CoolBlue from "@assets/cool-blue.svg";
import Party from "@assets/icons/party.svg";
import Sign from "@assets/icons/sign.svg";

const HeaderOverviewData = {
  title: "Black Friday 2019 blog",
  backButtonText: "Alle blogs",
  backButtonURL: "/",
  items: [
    {
      title: "De beste Moederdag cadeau deals",
      link: "/",
      image: Oasis
    },
    {
      title: "Black Friday wordt Black Week – deals beginnen eerder",
      link: "/",
      image: Oasis
    }
  ]
};

const FeaturedDeals = {
  title: "Onze beste deals",
  paddingTop: "72px",
  paddingBottom: "72px",
  bgcolor: "#f7f7f7",
  items: [
    {
      image: Oasis,
      sub_title: "Coolblue",
      title: "Apple AirPods 2 met draadloze oplaadcase",
      sale: "Bespaar 40% ",
      price: "€2.429",
      old_price: "€3.429",
      button_text: "Naar deals",
      button_link: "#"
    },
    {
      image: Oasis,
      sub_title: "Coolblue",
      title: "Apple AirPods 2 met draadloze oplaadcase",
      sale: "Bespaar 40% ",
      price: "€2.429",
      old_price: "€3.429",
      button_text: "Naar deals",
      button_link: "#"
    },
    {
      image: Oasis,
      sub_title: "Coolblue",
      title: "Apple AirPods 2 met draadloze oplaadcase",
      sale: "Bespaar 40% ",
      price: "€2.429",
      old_price: "€3.429",
      button_text: "Naar deals",
      button_link: "#"
    },
    {
      image: Oasis,
      sub_title: "Coolblue",
      title: "Apple AirPods 2 met draadloze oplaadcase",
      sale: "Bespaar 40% ",
      price: "€2.429",
      old_price: "€3.429",
      button_text: "Naar deals",
      button_link: "#"
    },
    {
      image: Oasis,
      sub_title: "Coolblue",
      title: "Apple AirPods 2 met draadloze oplaadcase",
      sale: "Bespaar 40% ",
      price: "€2.429",
      old_price: "€3.429",
      button_text: "Naar deals",
      button_link: "#"
    },
    {
      image: Oasis,
      sub_title: "Coolblue",
      title: "Apple AirPods 2 met draadloze oplaadcase",
      sale: "Bespaar 40% ",
      price: "€2.429",
      old_price: "€3.429",
      button_text: "Naar deals",
      button_link: "#"
    },
    {
      image: Oasis,
      sub_title: "Coolblue",
      title: "Apple AirPods 2 met draadloze oplaadcase",
      sale: "Bespaar 40% ",
      price: "€2.429",
      old_price: "€3.429",
      button_text: "Naar deals",
      button_link: "#"
    },
    {
      image: Oasis,
      sub_title: "Coolblue",
      title: "Apple AirPods 2 met draadloze oplaadcase",
      sale: "Bespaar 40% ",
      price: "€2.429",
      old_price: "€3.429",
      button_text: "Naar deals",
      button_link: "#"
    },
    {
      image: Oasis,
      sub_title: "Coolblue",
      title: "Apple AirPods 2 met draadloze oplaadcase",
      sale: "Bespaar 40% ",
      price: "€2.429",
      old_price: "€3.429",
      button_text: "Naar deals",
      button_link: "#"
    }
  ],
  buttonMore: [
    {
      title: "Er zijn meer interessante Apple deals!",
      image: Party,
      button_text: "Naar deals",
      button_link: "#"
    }
  ]
};

const CTASimple = {
  bgcolor: "#f7f7f7",
  paddingTop: "0",
  paddingBottom: "70px",
  title: "Op de hoogte blijven? Meld je dan aan voor onze nieuwsbrief!",
  buttonTitle: "Meld je aan",
  buttonLink: "#",
  icon: Sign
};
const BlogOverview = {
  bgcolor: "",
  paddingTop: "0",
  paddingBottom: "72px",
  sidebar: true,
  items: [
    {
      title: "Een greep uit onze deals",
      links: [
        {
          title: "Apple Airpods",
          link: "/",
          icon: Oasis
        },
        {
          title: "Apple iPhones",
          link: "/",
          icon: Oasis
        },
        {
          title: "Apple MacBooks",
          link: "/",
          icon: Oasis
        },
        {
          title: "Naar alle deals!",
          link: "/",
          icon: Party
        }
      ]
    },
    {
      title: "Black Friday Winkels",
      links: [
        {
          title: "Coolblue",
          link: "/",
          icon: CoolBlue
        },
        {
          title: "Alternate",
          link: "/",
          icon: CoolBlue
        },
        {
          title: "Beddenreus",
          link: "/",
          icon: CoolBlue
        },
        {
          title: "Naar alle Winkels!",
          link: "/",
          icon: Party
        }
      ]
    }
  ]
};

const PostsListDummy = [
  {
    image: Oasis,
    title: "Vijf voorspellingen van Black Friday Nederland",
    content:
      "Nog even en alles wordt weer uit de kast gehaald voor de beste Black Friday deals van 2018. Maar voordat het zo ver is, kijken we nog even terug... ",
    linkTitle: "Lees meer",
    link: "/blog-single"
  },
  {
    image: Oasis,
    title: "Top Black Friday deals van vorig jaar",
    content:
      "Nog even en alles wordt weer uit de kast gehaald voor de beste Black Friday deals van 2018. Maar voordat het zo ver is, kijken we nog even terug... ",
    linkTitle: "Lees meer",
    link: "/blog-single"
  },
  {
    image: Oasis,
    title: "Mis geen enkele Black Friday deal!",
    content:
      "Nog even en alles wordt weer uit de kast gehaald voor de beste Black Friday deals van 2018. Maar voordat het zo ver is, kijken we nog even terug... ",
    linkTitle: "Lees meer",
    link: "/blog-single"
  },
  {
    image: Oasis,
    title: "Wacht met shoppen tot Black Friday!",
    content:
      "Nog even en alles wordt weer uit de kast gehaald voor de beste Black Friday deals van 2018. Maar voordat het zo ver is, kijken we nog even terug... ",
    linkTitle: "Lees meer",
    link: "/blog-single"
  },
  {
    image: Oasis,
    title: "5 redenen om te wachten met shoppen tot Black Friday",
    content:
      "Nog even en alles wordt weer uit de kast gehaald voor de beste Black Friday deals van 2018. Maar voordat het zo ver is, kijken we nog even terug... ",
    linkTitle: "Lees meer",
    link: "/blog-single"
  }
];

export {
  BlogOverview,
  HeaderOverviewData,
  FeaturedDeals,
  CTASimple,
  PostsListDummy
};
