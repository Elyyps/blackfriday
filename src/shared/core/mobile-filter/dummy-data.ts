import { IMobileFilterItem } from "./mobile-filter-item";

export const mobileFilterDummyData: IMobileFilterItem[] = [
  {
    hasSearchBar: true,
    title: "Sorteren",

    selectedItems: ["Alphabetically"],
    isSingleSelection: true,
    items: ["Alphabetically", "Order"]
  },
  {
    hasSearchBar: true,
    title: "Status",
    searchBarPlaceholder: "Search in Status",
    selectedItems: [],
    items: ["Nu geldig", "Bijna verlopen", "Lorem Ipsum"]
  },
  {
    hasSearchBar: true,
    title: "Categorie",
    searchBarPlaceholder: "Search in Categories",
    selectedItems: [],
    items: ["Eletronics", "Fitness", "Music", "House and garden"]
  },
  {
    hasSearchBar: true,
    title: "Merk",
    searchBarPlaceholder: "Search in Brands",
    selectedItems: [],
    items: ["Nike", "Adidas", "Reebok", "New Balance"]
  }
];
