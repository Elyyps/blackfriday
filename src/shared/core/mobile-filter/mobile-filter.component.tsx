import React, { useState } from "react";
import styles from "./mobile-filter-component.module.scss";
import { FilterItemRow } from "@app/core/filter-item-row";
import { IconComponent } from "../icon";
import HandPointing from "@assets/icons/hand-pointing.svg";
import ChevronLeft from "@assets/icons/chevron-left.svg";
import FilterIcon from "@assets/icons/filter.svg";
import Cross from "@assets/icons/cross.svg";
import { Button } from "../button";
import { GenericPageFilterComponent } from "./pages/generic-page-filter";
import { IMobileFilterItem } from "./mobile-filter-item";
import { SingleFilterComponent } from "./pages/single-page-filter";
import { injectIntl, InjectedIntlProps } from "react-intl";

export interface IMobileFilterComponentProps {
  filterItems: IMobileFilterItem[];
  onClear: () => void;
  onFinish: (selectedItems: IMobileFilterSelectedItems[]) => void;
  totalStores: number;
}

export interface IMobileFilterSelectedItems {
  selectedItems: string[];
  title: string;
}

const component = (props: IMobileFilterComponentProps & InjectedIntlProps) => {
  const [currentFilterItem, setCurrentFilterItem] = useState<IMobileFilterItem | undefined>(undefined);
  const [currentFilterItems, setCurrentFilterItems] = useState<IMobileFilterItem[]>([]);
  const { filterItems, onClear } = props;
  const [isFilterOpened, setIsFilterOpened] = useState<boolean>(false);

  const onFinishHandler = () => {
    const selectedItems: IMobileFilterSelectedItems[] = filterItems.map(item => ({
      title: item.title,
      selectedItems: item.selectedItems
    }));
    props.onFinish(selectedItems);
    setIsFilterOpened(false);
  };

  const setSelectedItems = (filterItem: IMobileFilterItem, items: string[]) => {
    if (currentFilterItems && currentFilterItem) {
      currentFilterItems.forEach(item => {
        if (item.title === filterItem.title) {
          item.selectedItems = items;
        }
      });

      setCurrentFilterItems(currentFilterItems);
    }
  };

  const onClearHandler = () => {
    if (currentFilterItem) {
      const newItem: IMobileFilterItem = {
        title: currentFilterItem.title,
        items: currentFilterItem.items,
        selectedItems: [],
        searchBarPlaceholder: currentFilterItem.searchBarPlaceholder,
        hasSearchBar: currentFilterItem.hasSearchBar
      };
      setCurrentFilterItem(newItem);
      setSelectedItems(currentFilterItem, []);
    } else {
      onClear();
    }
  };
  React.useEffect(() => {
    setCurrentFilterItems(filterItems);
  }, [filterItems]);

  React.useEffect(() => {
    if (isFilterOpened) {
      document.documentElement.style.overflow = "hidden";
      document.body.style.overflowY = "scroll";
    } else {
      document.documentElement.style.overflow = "auto";
      document.body.style.overflow = "auto";
    }

    return () => {
      document.documentElement.style.overflow = "auto";
      document.body.style.overflow = "auto";
    };
  }, [isFilterOpened]);

  const onWindowResize = () => {
    const w = window;
    const d = document;
    const documentElement = d.documentElement;
    const body = d.getElementsByTagName("body")[0];
    const width = w.innerWidth || documentElement.clientWidth || body.clientWidth;
    const mobileBreakpoint = 769;
    if (width >= mobileBreakpoint) {
      document.documentElement.style.overflow = "auto";
      document.body.style.overflow = "auto";
    }
  };

  React.useEffect(() => {
    window.addEventListener("resize", onWindowResize);

    return () => window.removeEventListener("resize", onWindowResize);
  }, []);

  return !isFilterOpened ? (
    <div
      role="button"
      onClick={() => {
        setIsFilterOpened(true);
      }}
      className={styles["filter-bar"]}
    >
      {props.intl.formatMessage({ id: "filter-bar-label" })}
      <IconComponent icon={FilterIcon} size={"15px"} />
    </div>
  ) : (
    <div className={styles["mobile-filter"]}>
      {currentFilterItem ? (
        <div className={styles["mobile-filter__header"]}>
          <span
            role="button"
            className={styles["mobile-filter__header__nav-button"]}
            onClick={() => {
              setCurrentFilterItem(undefined);
            }}
          >
            <IconComponent strokeColor="#fff" icon={ChevronLeft} size={"12px"} />
          </span>
          <span className={styles["mobile-filter__header__title"]}>
            <span>{currentFilterItem.title}</span>
          </span>
          {!currentFilterItem.isSingleSelection && (
            <a role="button" onClick={onClearHandler} className={styles["mobile-filter__header__clear"]}>
              {props.intl.formatMessage({ id: "mobile-filter-clear-filter" })}
            </a>
          )}
        </div>
      ) : (
        <div className={styles["mobile-filter__header"]}>
          <span
            role="button"
            className={styles["mobile-filter__header__nav-button"]}
            onClick={() => {
              setIsFilterOpened(false);
            }}
          >
            <IconComponent strokeColor="#fff" icon={Cross} size={"12px"} />
          </span>

          <div className={styles["mobile-filter__header__title"]}>
            <span>{props.intl.formatMessage({ id: "mobile-filter-title" })}</span>
            <IconComponent icon={HandPointing} size={"16px"} />
          </div>
          <a
            role="button"
            onClick={() => {
              onClear();
            }}
            className={styles["mobile-filter__header__clear"]}
          >
            {props.intl.formatMessage({ id: "mobile-filter-clear-filter" })}
          </a>
        </div>
      )}

      {!currentFilterItem ? (
        currentFilterItems.map((item, key) => (
          <FilterItemRow
            key={key}
            items={item.selectedItems}
            onClick={() => {
              setCurrentFilterItem(item);
            }}
            title={item.title}
          />
        ))
      ) : (
        <div className={styles["mobile-filter__list"]}>
          {currentFilterItem.isSingleSelection ? (
            <SingleFilterComponent
              hideSearchBar={!currentFilterItem.hasSearchBar}
              items={currentFilterItem.items}
              selectedItem={currentFilterItem.selectedItems[0]}
              placeholder={currentFilterItem.searchBarPlaceholder}
              setCheckedItem={(selectedItem: string) => {
                const selectedItemArray = [];
                selectedItemArray.push(selectedItem);
                setSelectedItems(currentFilterItem, selectedItemArray);
                setCurrentFilterItem(undefined);
              }}
            />
          ) : (
            <GenericPageFilterComponent
              hideSearchBar={!currentFilterItem.hasSearchBar}
              setCheckedItems={(selectedItems: string[]) => {
                setSelectedItems(currentFilterItem, selectedItems);
              }}
              placeholder={currentFilterItem.searchBarPlaceholder}
              filterItem={currentFilterItem}
            />
          )}
        </div>
      )}
      <div className={styles["mobile-filter__footer"]}>
        <Button
          variant="primary-brand"
          fullWidth
          onClick={onFinishHandler}
          title={props.intl.formatMessage({ id: "mobile-filter-button" })}
        />
      </div>
    </div>
  );
};

const MobileFilterComponent = injectIntl(component);
export { MobileFilterComponent };
