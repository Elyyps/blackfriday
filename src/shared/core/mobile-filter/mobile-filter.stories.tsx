import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { MobileFilterComponent } from "./mobile-filter.component";
import { mobileFilterDummyData } from "./dummy-data";
import { breakPointDesktop } from "@app/util/detect-view";
import { ViewType } from "@app/stores/settings";

storiesOf("MobileFilter", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => (
    <MobileFilterComponent
      onFilterChange={jest.fn}
      screenSize={{ viewType: ViewType.Desktop, breakpointPixels: breakPointDesktop }}
      totalStores={0}
      onClear={jest.fn}
      onFinish={jest.fn}
      filterItems={mobileFilterDummyData}
    />
  ));
