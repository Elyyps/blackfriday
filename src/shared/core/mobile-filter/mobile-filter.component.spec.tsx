import React from "react";
import { shallow } from "enzyme";
import { MobileFilterComponent } from "./mobile-filter.component";
import { mobileFilterDummyData } from "./dummy-data";

describe("[MobileFilter]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(
      <MobileFilterComponent totalStores={0} onClear={jest.fn} onClose={jest.fn} filterItems={mobileFilterDummyData} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
