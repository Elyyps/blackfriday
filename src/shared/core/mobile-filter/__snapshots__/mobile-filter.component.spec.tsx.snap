// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[MobileFilter] should render component without crashing 1`] = `
<component
  filterItems={
    Array [
      Object {
        "hasSearchBar": true,
        "isSingleSelection": true,
        "items": Array [
          "Alphabetically",
          "Order",
        ],
        "selectedItems": Array [
          "Alphabetically",
        ],
        "title": "Sorteren",
      },
      Object {
        "hasSearchBar": true,
        "items": Array [
          "Nu geldig",
          "Bijna verlopen",
          "Lorem Ipsum",
        ],
        "searchBarPlaceholder": "Search in Status",
        "selectedItems": Array [],
        "title": "Status",
      },
      Object {
        "hasSearchBar": true,
        "items": Array [
          "Eletronics",
          "Fitness",
          "Music",
          "House and garden",
        ],
        "searchBarPlaceholder": "Search in Categories",
        "selectedItems": Array [],
        "title": "Categorie",
      },
      Object {
        "hasSearchBar": true,
        "items": Array [
          "Nike",
          "Adidas",
          "Reebok",
          "New Balance",
        ],
        "searchBarPlaceholder": "Search in Brands",
        "selectedItems": Array [],
        "title": "Merk",
      },
    ]
  }
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  onClear={[Function]}
  onFinish={[Function]}
  screenSize={
    Object {
      "breakpointPixels": 960,
      "viewType": 3,
    }
  }
  totalStores={0}
/>
`;
